File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/SalaryGradeForm.ts
import { BooleanEditor, DecimalEditor, initFormType, PrefixedContext, StringEditor, TextAreaEditor } from "@serenity-is/corelib";

export interface SalaryGradeForm {
    GradeName: StringEditor;
    GradeCode: StringEditor;
    MinSalary: DecimalEditor;
    MaxSalary: DecimalEditor;
    Description: TextAreaEditor;
    IsActive: BooleanEditor;
}

export class SalaryGradeForm extends PrefixedContext {
    static readonly formKey = 'Operations.SalaryGrade';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SalaryGradeForm.init) {
            SalaryGradeForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = TextAreaEditor;
            var w3 = BooleanEditor;

            initFormType(SalaryGradeForm, [
                'GradeName', w0,
                'GradeCode', w0,
                'MinSalary', w1,
                'MaxSalary', w1,
                'Description', w2,
                'IsActive', w3
            ]);
        }
    }
}

File: HRMS/HRMS.Web/Modules/Operations/SalaryGrade/SalaryGradeForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMS.Operations.Forms;

[FormScript("Operations.SalaryGrade")]
[BasedOnRow(typeof(SalaryGradeRow), CheckNames = true)]
public class SalaryGradeForm
{
    public string GradeName { get; set; }
    public string GradeCode { get; set; }
    public decimal MinSalary { get; set; }
    public decimal MaxSalary { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }
    public bool IsActive { get; set; }
}
